tests/Integration/Unit/UnitTest.php

```diff
@@ -3,39 +3,26 @@
 namespace SystemCtl\Test\Integration\Unit;
 
 use PHPUnit\Framework\TestCase;
-use Symfony\Component\Process\Process;
-use Symfony\Component\Process\ProcessBuilder;
+use Prophecy\Argument;
+use Prophecy\Prophecy\ObjectProphecy;
+use SystemCtl\Command\CommandDispatcherInterface;
+use SystemCtl\Command\CommandInterface;
 use SystemCtl\Exception\CommandFailedException;
 use SystemCtl\SystemCtl;
-use SystemCtl\Unit\Service;
 
 class UnitTest extends TestCase
 {
     public function testServiceCommandsIfProcessIsSuccessfulShouldReturnTrue()
     {
-        $process = $this->getMockBuilder(Process::class)
-            ->disableOriginalConstructor()
-            ->setMethods(['isSuccessful'])
-            ->getMock();
+        $command = $this->prophesize(CommandInterface::class);
+        $command->isSuccessful()->willReturn(true);
 
-        $processBuilder = $this->getMockBuilder(ProcessBuilder::class)
-            ->disableOriginalConstructor()
-            ->setMethods(['getProcess'])
-            ->getMock();
-
-        $processBuilder->method('getProcess')->willReturn($process);
-
-        /** @var \PHPUnit_Framework_MockObject_MockObject|SystemCtl $systemctl */
-        $systemctl = $this->getMockBuilder(SystemCtl::class)
-            ->setMethods(['getProcessBuilder'])
-            ->getMock();
-
-        $systemctl->method('getProcessBuilder')->willReturn($processBuilder);
+        $commandDispatcher = $this->createCommandDispatcherStub();
+        $commandDispatcher->dispatch(Argument::cetera())->willReturn($command);
 
+        $systemctl = (new SystemCtl())->setCommandDispatcher($commandDispatcher->reveal());
         $service = $systemctl->getService('AwesomeService');
```

Comment:
You do not instanciate your Unit via an external service. You simply do `new XXX()` in unit tests ;)

- [ ] Fix tests everywhere